### PR TITLE
Support JDK 11 and JDK 17 Compilation

### PR DESCRIPTION
**Pull request by YogyaGamage**
### Purpose
Support JDK 11 and JDK 17 compilation.

After this PR is merged to the main branch, it is possible to build the branch on JDK 8 as well.

### Related Issue
https://github.com/wso2/product-is/issues/14224

### CODE REVIEW

**Comment by piraveena** [comment]
will this support jdk8 along with jdk11 support?

**Comment by YogyaGamage** [comment]
> will this support jdk8 along with jdk11 support?

Yes

**Comment by YogyaGamage** [comment]
> @YogyaGamage After merging the PR do we need to update the README file?

The existing README file does not need to be changed as it is about implementing API and does not contain any information regarding compilation support, but they can be added if needed.

File: pom.xml [hunk 1]
```diff
@@ -239,6 +239,12 @@
                 <artifactId>commons-lang</artifactId>
                 <version>${commons-lang.wso2.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.projectlombok</groupId>
+                <artifactId>lombok</artifactId>
+                <version>${lombok.version}</version>
+                <scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (YogyaGamage) [reply pom.xml:246]: > 1. Why this dependency is needed?

It is needed to fix the error _"java.lang.IllegalAccessError: class lombok.javac.apt.LombokProcessor cannot access class com.sun.tools.javac.processing.JavacProcessingEnvironment"_ as mentioned in https://stackoverflow.com/questions/66801256/java-lang-illegalaccesserror-class-lombok-javac-apt-lombokprocessor-cannot-acce. 
The older versions of lombok seems to have the issue, so it needs to be updated. 
AFAIK the dependency comes from https://github.com/wso2-extensions/identity-local-auth-fido/blob/master/pom.xml. It has the version 1.18.8 which causes the issue.

> 2. Will it be packed with the product?

An older version of the dependency already exists (not in this repo, but in fido which is a dependency in this repo). 

> 3. If so, is it apache compliance dependency?

Yes, it has MIT license which is compatible with Apache.

> 4. Its added to root pom only, is there no specific components that need this dependency?

No, it is only needed in the root pom.

File: .github/workflows/pr-builder.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # This workflow will build the project on pull requests with tests
 # Uses:
 #   OS: ubuntu-latest
-#   JDK: Adopt JDK 8
+#   JDK: Adopt JDK 11 and Adopt JDK 17
```
Comment (piraveena) [inline .github/workflows/pr-builder.yml:4]: shall we update those readme saying, the component can be compiled with jdk8,11 and 17

**Review by piraveena — APPROVED** [review]

**Pull request merged**